src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -2314,7 +2317,7 @@ private static void EmitDereference(ILGenerator il, Type type)
                 var method = expr.Method;
                 if (method != null && method.Name != "op_Implicit" && method.Name != "op_Explicit")
                 {
-                    if (!TryEmit(opExpr, paramExprs, il, ref closure, parent & ~ParentFlags.IgnoreResult | ParentFlags.InstanceCall, 0))
+                    if (!TryEmit(opExpr, paramExprs, il, ref closure, parent & ~ParentFlags.IgnoreResult | ParentFlags.InstanceCall, -1))

```

Comment:
![image](https://user-images.githubusercontent.com/2588062/75392126-aeb6dd00-58eb-11ea-9d2f-8fd4c3571405.png)

This `0`  caused my other fix to start emitting by reference loads since this tells that first argument should be passed by reference from what I understand. And at least in unit tests this code was calling methods that take arguments by value so tests started to fail.

Reply:
Huh, ok seems a legit fix from the first sight.